webrtc-stats.html

```diff
@@ -2898,8 +2905,8 @@ <h3>
         </p>
         <div>
           <pre class="idl">dictionary RTCRtpContributingSourceStats : RTCStats {
-             unsigned long contributorSsrc;
-             DOMString     inboundRtpStreamId;
+             required unsigned long contributorSsrc;
+             required DOMString     inboundRtpStreamId;
              unsigned long packetsContributedTo;
```

Comment:
Here's an example of a counter that might not be very important, but it always has a defined value if implemented (unlike framesDecoded which is only valid for video and thus has a sensible definition of undefined: the audio stream case).

Opinion: if the only meaning of "undefined" for a metric is "it hasn't been implemented" then I think is should be required, otherwise it is like saying "you can implemented this as undefined by not implementing it"... that sounds self-contradictory.

Reply:
Seems like a reasonable argument. However, it applies to all counters.
The counterargument is for the case where an implementor sees value in knowing what ssrcs exist as CSRCs, but doesn't care about the packet value (or finds it too hard to implement correctly) - that implementation is either conformant or non-conformant to this spec; if we make the counter mandatory, it's non-conformant even in the absence of a conformance statement anywhere else.

I think I'm arguing myself into leaving it non-required.

Reply:
OK. Am I correct in thinking that this implies that `required` means "the subset of mandatory to implement stats that can't be undefined if implemented"?